File: app/src/main/java/org/deiverbum/app/core/module/FirebaseModule.java
package org.deiverbum.app.core.module;

import com.google.firebase.firestore.FirebaseFirestore;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;


@Module
@InstallIn(SingletonComponent.class)
        public abstract class FirebaseModule {

        @Provides
        @Singleton
        static FirebaseFirestore provideFirebaseInstance(){
                return FirebaseFirestore.getInstance();
        }

        }

File: app/src/main/java/org/deiverbum/app/core/module/FirebaseModule.kt
package org.deiverbum.app.core.module

import com.google.firebase.firestore.FirebaseFirestore
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
object FirebaseModule {
    @JvmStatic
    @Provides
    @Singleton
    fun provideFirebaseInstance(): FirebaseFirestore {
        return FirebaseFirestore.getInstance()
    }
}